src/Cli/dotnet/CommonOptions.cs

```diff
@@ -109,25 +111,22 @@ public static CliArgument<string> DefaultToCurrentDirectory(this CliArgument<str
 
         public static CliOption<bool> NoRestoreOption = new ForwardedOption<bool>("--no-restore")
         {
-            Description = CommonLocalizableStrings.NoRestoreDescription
+            Description = CommonLocalizableStrings.NoRestoreDescription,
+            Arity = ArgumentArity.Zero
         }.ForwardAs("-restore:false");
 
         public static CliOption<bool> InteractiveMsBuildForwardOption =
             new ForwardedOption<bool>("--interactive")
             {
-                Description = CommonLocalizableStrings.CommandInteractiveOptionDescription
+                Description = CommonLocalizableStrings.CommandInteractiveOptionDescription,
+                Arity = ArgumentArity.Zero
             }.ForwardAs("-property:NuGetInteractive=true");
 
-        public static CliOption<bool> InteractiveOption =
-            new("--interactive")
-            {
-                Description = CommonLocalizableStrings.CommandInteractiveOptionDescription
-            };
-
         public static CliOption<bool> DisableBuildServersOption =
             new ForwardedOption<bool>("--disable-build-servers")
             {
-                Description = CommonLocalizableStrings.DisableBuildServersOptionDescription
+                Description = CommonLocalizableStrings.DisableBuildServersOptionDescription,
+                Arity = ArgumentArity.Zero
```

Comment:
Does adding Arity = ArgumentArity.Zero in all these places actually do anything? Prevent you from saying --disable-build-servers:true? Would we want to prevent that?

Reply:
This is where things get interesting. You're right that we're technically preventing this syntax, but this is a forwarded option, one whose whole purpose is to be forwarded as MSBuild properties. This means we have to look at the forwarding function to see if the flag value even matters - and in this case the forwarding function is hard-coded to not look at the boolean's value at all. In this case, the currently-chosen semantic is a flag, not an option with an argument. Knowing this makes the completions generation better in some cases, so I've nailed down the behavior in some scenarios like this.